### ISSUE TITLE
`syntax` analog for IDE

### ISSUE THREAD

**Issue by andreyvolokitin** [stylelint/stylelint#3211]
We can use `syntax` to make stylelint recognize preferred syntax for parsing. But I can't find a similar option for IDEs and editors. I want to use SASS feature (inline comments) within regular .css files handled by postcss. I managed to make it work with ```postcss-scss``` and stylelint `syntax` so my build is passing without errors. But inside IDE (webstorm) I get errors because stylelint still parses .css as CSS. I thought that maybe there is the same `syntax` option that you can put inside `stylelint.config.js` so that editor can rely on that, but that is not the case. How should this be handled and is there a solution for this? 

> Which rule, if any, is this issue related to?

None

> What CSS is needed to reproduce this issue?

```css
// Inline SASS comment
```

> What stylelint configuration is needed to reproduce this issue?

Any config I guess

> Which version of stylelint are you using?

`9.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Webstorm-integrated stylelint with `stylelint.config.js` 

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS inline comments

**Comment by ntwb** [comment]
I'm not familiar with Webstorm, that said after a quick search I found this:
• Via https://github.com/Jardinero/stylelint-plugin/blob/develop/resources/META-INF/plugin.xml#L28
• `<externalAnnotator language="SCSS" implementationClass="com.jokerzoid.intellij.plugin.stylelint.StylelintAnnotator" />`

Is `StylelintAnnotator` another plugin available for Webstorm by chance?

**Comment by andreyvolokitin** [comment]
No, there is no `StylelintAnnotator` in plugins list for webstorm (in fact `stylelint-plugin` is the only plugin available)

**Comment by andreyvolokitin** [comment]
I think this definitely could be implemented by IDE/plugin, but maybe with somewhat low probability of happening. What I thought is — could this possibly be instructed by stylelint configuration, so that all IDEs/editors could rely on that? Currently, I suppose they switch parsing depending on file extension. But again I guess this may be quite an edge case to be implemented. 

If I simply start using .scss file extension then:
- I will need another processing stage (transpiling to regular CSS)
- more importantly, I'm worried that my current setup (postcss with plugins for variables/mixins/nesting/etc) will somehow interfere with SCSS syntax. Transpiling postcss before SCSS maybe could resolve this... Not sure about IDE syntax highlighting

**Comment by jeddy3** [comment]
>  I want to use SASS feature (inline comments) within regular .css files handled by postcss. I managed to make it work with postcss-scss and stylelint syntax so my build is passing without errors.

I'm a little confused by this use case. Am I right in thinking you're writing SCSS but using the `.css`, rather than the `.scss` extension? The happy path is to use the `.scss` extension if you're writing SCSS.

> Currently, I suppose they switch parsing depending on file extension. 

This is how stylelint does it too.

> more importantly, I'm worried that my current setup (postcss with plugins for variables/mixins/nesting/etc) will somehow interfere with SCSS syntax.

Am I right in thinking you're using PostCSS plugins to add language extensions like variables, mixins and nestings _but also_ using SCSS syntax so that you can use the single line `//` comments? I'd question the value of complicating your syntax parsing just for `//` comments. If you really have to stick with SCSS syntax, though, then using the `.scss` file extension is going to offer you the most compatibility with tools.

**Comment by Arcanemagus** [comment]
> > Currently, I suppose they switch parsing depending on file extension. 
> 
> This is how stylelint does it too.

Just to note, the Atom package follows whatever the editor thinks the file is, which _defaults_ based on the file extension, but if you tell Atom to treat a CSS file as SCSS, it will work with it as if it was a SCSS file. So this is more of an implementation detail of the specific Editor/Plugin you are using.

**Comment by andreyvolokitin** [comment]
> Am I right in thinking you're writing SCSS but using the .css, rather than the .scss extension?

I actually don't want to use SCSS, and I am not using/writing it overall, except inline comments, so all my CSS is regular CSS (except quite a few inline comments) within .css files. I only want this one single SCSS feature — inline comments, it is really convenient to have disappearing comments for stuff that disappears after compilation: variables, mixins, etc. But because it is an SCSS feature I am forced to use `postcss-scss` and stylelint `syntax: 'scss'` option.

> Am I right in thinking you're using PostCSS plugins to add language extensions like variables, mixins and nestings but also using SCSS syntax so that you can use the single line // comments?

Yes, I am using SCSS parsing with `postcss-scss` and stylelint `syntax: 'scss'` option only for one SCSS feature — inline comments

---------
All in all,  I considered giving up inlines comments feature but I am still trying to find a way. I also considered converting to .scss, but I think this is the least attractive option.

**Comment by andreyvolokitin** [comment]
@Arcanemagus, I think webstorm has something similar when you can set different file types to a file extension. Gotta check this, but I don't know if stylelint will follow along with IDE to also treat a CSS file as SCSS

**Comment by jeddy3** [comment]
> Gotta check this, but I don't know if stylelint will follow along with IDE to also treat a CSS file as SCSS

stylelint is usually told, via the [syntax option](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/node-api.md#syntax), what syntax to use _by the editor (or extension)_ e.g. this is how the [atom extension](https://github.com/AtomLinter/linter-stylelint/blob/master/lib/index.js#L106-L112) does it.

So, as @Arcanemagus said, I believe this is also more of an implementation detail of the specific Editor/Plugin you are using.

**Comment by andreyvolokitin** [comment]
OKay, you can't assign multiple file types to a single file extension in webstorm, `.css` can be either SCSS or CSS, and across all projects at once. Stylelint errors were resolved though, thanks! I will think if it worth to keep it that way
